File: code_blocks/integrations/third-party-integrations/firebase-integration_2.kt
import android.app.Application
import com.google.firebase.analytics.ktx.analytics
import com.google.firebase.ktx.Firebase
import com.revenuecat.purchases.Purchases

class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        Purchases.configure(this, "public_sdk_key")
        Firebase.analytics.appInstanceId.addOnCompleteListener { task ->
            if (task.isSuccessful) {
                task.result?.let { id ->
                    Purchases.sharedInstance.attribution.setFirebaseAppInstanceId(id)
                }
            }
        }
    }
}

File: code_blocks/integrations/third-party-integrations/firebase-integration_2.dart
import 'package:firebase_analytics/firebase_analytics.dart';
import 'package:purchases_flutter/purchases_flutter.dart';

Future<void> initPlatformState() async {
  await Purchases.configure(PurchasesConfiguration('public_sdk_key'));
  final instanceId = await FirebaseAnalytics.instance.appInstanceId;
  if (instanceId != null) {
    await Purchases.setFirebaseAppInstanceId(instanceId);
  }
}
